packages/suite/src/utils/suite/homescreen.ts

```diff
@@ -186,48 +186,65 @@ export const elementToImageData = (element: HTMLImageElement, width: number, hei
-        throw new Error('Not a correct height.');
-    }
-    if (origImage.width !== width) {
-        throw new Error('Not a correct width.');
-    }
-
     const imageData = elementToImageData(origImage, width, height);
 
     if (model === 1) {
-        range(imageData.height).forEach((j: number) => {
-            range(imageData.width).forEach(i => {
-                const index = j * 4 * imageData.width + i * 4;
-                const red = imageData.data[index];
-                const green = imageData.data[index + 1];
-                const blue = imageData.data[index + 2];
-                const alpha = imageData.data[index + 3];
-                if (alpha !== 255) {
-                    throw new Error('Unexpected alpha.');
-                }
-                let good = false;
-                if (red === 0 && green === 0 && blue === 0) {
-                    good = true;
-                }
-                if (red === 255 && green === 255 && blue === 255) {
-                    good = true;
-                }
-                if (!good) {
-                    throw new Error(`wrong color combination ${red} ${green} ${blue}.`);
-                }
+        try {
+            range(imageData.height).forEach((j: number) => {
+                range(imageData.width).forEach(i => {
+                    const index = j * 4 * imageData.width + i * 4;
+                    const red = imageData.data[index];
+                    const green = imageData.data[index + 1];
+                    const blue = imageData.data[index + 2];
+                    const alpha = imageData.data[index + 3];
+                    if (alpha !== 255) {
+                        throw new Error(ImageValidationError.UnexpectedAlpha);
+                    }
+                    const isBlack = red === 0 && green === 0 && blue === 0;
+                    const isWhite = red === 255 && green === 255 && blue === 255;
+
+                    if (!isBlack && !isWhite) {
+                        throw new Error(ImageValidationError.InvalidColorCombination);
+                    }
+                });
             });
-        });
+        } catch (error) {
+            return error.message;
+        }
     }
 };
 
-export const check = (file: File, model: number) =>
-    fileToDataUrl(file)
-        .then((url: string) => dataUrlToImage(url))
-        .then(image => checkImage(image, model));
+export const validateImageDimensions = (origImage: HTMLImageElement, model: number) => {
+    const height = getHeight(model);
+    const width = getWidth(model);
+    if (origImage.height !== height) {
+        return ImageValidationError.InvalidHeight;
+    }
+    if (origImage.width !== width) {
+        return ImageValidationError.InvalidWidth;
+    }
+};
+
+export const validateImageFormat = (dataUrl: string) =>
+    !!dataUrl && supportedDataUrlRE.test(dataUrl) ? undefined : ImageValidationError.InvalidFormat;
+
+export const validate = (dataUrl: string, model: number) =>
```

Comment:
Please use some discriminated union for model instead of number, that is quite cryptic what it could be. Something like this `type ModelNumber = 1 | 2` would be really nice

Reply:
I agree, this is very confusing across the code base. 
* There are some places where we have `trezorModel: 1 | 2;` as a prop attribute
* There are other places where we use strings "1" and "T" like in `packages/suite/src/utils/suite/device.ts` :
```javascript
export const getDeviceModel = (device: TrezorDevice) => {
    const { features } = device;
    return features && features.major_version > 1 ? 'T' : '1';
};
```

I have the feeling that having some global type for `trezorModel` that helps us to be consistent would be useful.

Reply:
Maybe something to bring to the code improvement session.

Reply:
@karliatto good point! 

For now I would recommend start here then we can use this type in other places.

Reply:
I agree, I just don't want to refactor it everywhere inside this PR, but it would be really nice to unify this 👍 

for now, I did only this local small improvement: c26a19c9524aada4045628d1305c9bb6eb82941e

Reply:
nice